lib/data/models/token_contract_asset.dart

```diff
@@ -0,0 +1,23 @@
+import 'package:app/data/models/network_type.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'token_contract_asset.freezed.dart';
+
+part 'token_contract_asset.g.dart';
+
+@freezed
+class TokenContractAsset with _$TokenContractAsset {
+  const factory TokenContractAsset({
+    required String name,
+    required String symbol,
+    required int decimals,
```

Comment:
I think that its better to put Currency right here. Because we already have name, symbol and decimals.

Reply:
It's not the same, because Currency do not contains decimals and symbols. Currency contains information only about price changes

Reply:
If ur talking about money2 lib, so it's incompatible, see below